### ISSUE TITLE
Lab1: Comments

### ISSUE THREAD

**Issue by ringavirda** [luvremak/programming-fundamentals#1]
### 1st assignment
- Where did that math come from? It was supposed to be simple increment operation.
- In the second implementation you go through all of the object keys and apply math without any type checks. What would happen if there were fields that are strings, objects, callables?

### 2nd assignment
- You use a forest of if-else statements to determine the type of next array element, which is not necessary since type names are in themselves strings and can be used as keys. All of this logic can be replaced with simple one line assignment.
- Dynamic implementation suffers from this decision even more, since you added guard statements, which are another level of if-else. I'm usually not opposed to the solutions that include a lot of logic, but here it's simply excessive and unnecessary.

**Comment by luvremak** [comment]
Thank you for your feedback. I have updated the code and fixed the issues you mentioned. 

 "data-types-2" is now more compact and simple, free of an excessive amount of if-else statements. I took a completely different approach, and the program output remains the same. 
I've added type checks to the scalar-type exercises and have simplified them both to the maximum extent using ternary operators.

The math expression I used is just from a textbook. I wanted to make it more interesting, but I didn't mean to make it too complex. Anyway, I have changed it.
